### PR TITLE
Add attention mask for chemberta

### LINKED ISSUE #3597: Pass attribute mask for chemberta model
Since we are padding tokens for masked language model training here,
https://github.com/deepchem/deepchem/blob/e815ffb5191520e078678b3183546c6042413507/deepchem/models/torch_models/hf_models.py#L219-L220
we should pass `attention_mask` to the model.

Steps to reproduce warning from `transformers` library:
```py
from deepchem.models import Chemberta
imort deepchem as dc

task, dataset, transformer = dc.molnet.load_delaney(featurizer=dc.feat.DummyFeaturizer(), reload=True)
train, test, valid = dataset

model = Chemberta(task='mlm', model_dir='pretrain-dir')
model.fit(train, nb_epoch=1, checkpoint_interval=1)
```
It will show a warning as
```
We strongly recommend passing in an `attention_mask` since your input_ids may be padded. See https://huggingface.co/docs/transformers/troubleshooting#incorrect-output-when-padding-tokens-arent-masked.
```

### PR DESCRIPTION
**Pull request by arunppsg**
Fix #3597

Summary of changes:
- pass attention mask for mlm
- added more user arguments for grover model

## Type of change

Please check the option that is related to your PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by arunppsg** [comment]
Note: The failure in code-formatting ci is not related to changes in this pull request.

**Review by rbharath — COMMENTED** [review]
@arunppsg Can you rebase? We just fixed jax CI so we shoudl see a lot more tests passing. I will do final review after

**Comment by arunppsg** [comment]
@rbharath rebased it

**Review by rbharath — APPROVED** [review]
LGTM
